### PR TITLE
Add in a new item category & zone: TOOL_MAGAZINE, and various category shuffling of items

### LINKED ISSUE #67607: The item categories of tanks (as in containers) are inconsistent and follows no discernable pattern.
### Describe the bug

Different categories for tanks in the game:

* pressurized tank WEAPONS
* pressurized fuel tank MAGAZINES
* propane tank TOOLS
* scuba tank OTHER
* re-breather oxygen tank is SPARE PARTS
* metal tank a CONTAINERS or whatever category it's content is filled 
* tank top CLOTHING (last one was a joke)

It's hard to find the tanks when you sort your zone by category. 

I wasn't sure whether to categorize this as a suggestion or bug report, but having pressurized tank categorized as WEAPONS can't be intended.

### Attach save file

N/A

### Steps to reproduce

N/A

### Expected behavior

Either:

1. we have everything that's used to reload stuff as MAGAZINES (or create a new category called TOOL MAGAZINES or similar),  or 2. we sort into to where the stuff they reload go, e.g. re-breather oxygen tank as CLOTHES as it reloads the re-breeding mask. 

General metal tanks or pressurized tanks without any associated tool can be containers and take the category of their content.

### Screenshots

image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.14393.6085 (1607)
- Game Version: 2852530 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda]
]

### Additional context

Also, for welding tools:

* Acetylene torch is loaded with a magazine (welding tank).
* Propane torch is loaded with a tool (propane tank).
* Arc welder is loaded with a spare part (battery) or a vehicle part (car battery or storage battery)

Batteries should maybe also be in the same category as tanks, whatever that may be.

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
features "magazine tools now have their own item category and zone"

#### Purpose of change
Fixes #67607
it's a mess otherwise, tool magazines are scattered everywhere and need to be in one category

#### Describe the solution
new item category and zone, renaming and recategorizing various items

#### Describe alternatives you've considered
NA

#### Testing
made sure zone and category works, check additional context

#### Additional context
image
261798736-90e75638-a32d-4531-a4ff-60d4bcf28f6d

### CODE REVIEW

**Comment by ampersand55** [comment]
It would be nice with some naming conventions. Like if you have a tank you would know it would be a container, and if you had a canister or cartridge you would know it was a tool mag. 

Some more tanks and canisters not mentioned in the issue:

butane tank  (TOOLS)

hydrogen tank  (TOOLS)

goo canister  (TOOLS)

methanol fuel cartridge (SPARE PARTS)

nanomaterial canister (CHEMICAL STUFF)

**Comment by ampersand55** [comment]
As it's described as a canister and you use it to reload what is essentially a tool. I think it should be a TOOL_MAGAZINE.

As far as I can tell, all other CHEMICAL STUFF are just pure chemicals.

File: data/json/items/tool/med.json [hunk 1]
```diff
@@ -322,7 +324,8 @@
   {
     "id": "oxygen_tank",
     "type": "TOOL",
-    "name": { "str": "oxygen tank" },
+    "name": { "str": "oxygen delivery kit" },
+    "category": "tools",
```
Comment (Karol1223) [inline data/json/items/tool/med.json:328]: I don't think it needs a category of ``tools`` specified if it's already ``type: TOOL``. Could you check that? I'm positive this is redundant.

File: data/json/items/tool/med.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "type": "TOOL",
     "name": { "str": "anesthesia kit" },
     "description": "A kit for inducing anesthesia for surgery, containing specialized canisters with a variety of powerful hypnotic, analgesic, and stimulative drugs.  It's intended for use in specialized medical equipment, and can't be administered manually.",
+    "category": "drugs",
```
Comment (Karol1223) [inline data/json/items/tool/med.json:38]: I'm not convinced on this. In itself this is just a container, not a drug. It's contents would be drugs, it in itself wouldn't.

File: data/json/items/tool/science.json [hunk 3]
```diff
@@ -605,6 +612,7 @@
     "type": "TOOL",
     "name": { "str": "hydrogen tank" },
     "description": "A tank of compressed hydrogen gas.  If you need to make water from scratch, or lift a zeppelin, it could come in handy.",
+    "category": "container",
```
Comment (Karol1223) [inline data/json/items/tool/science.json:615]: It doesn't function as a container. Not sure where it'd fit best, but this is realistically either spare parts or chems. It's a glorified resource item for hydrogen.

File: data/json/items/tool/science.json [hunk 4]
```diff
@@ -668,6 +677,7 @@
     "type": "TOOL",
     "name": { "str": "nitrogen tank" },
     "description": "A tank of compressed nitrogen gas.  Nitrogen is useful for its lack of reactivity.  Don't try to breathe it.",
+    "category": "container",
```
Comment (Karol1223) [inline data/json/items/tool/science.json:680]: Same as hydrogen tank

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by Karol1223** [comment]
> Do worn containers count as clothing? (hip flask and canteen for example)

My personal criteria I've been going through was if it's wearble then it's fine as clothing, but if it's "clothing" that uses our primitive dual wielding (laundry baskets, plastic shopping bags, briefcases) it would be containers. Btw I think laundry baskets and briefcases are indeed clothing still if you wanna touch that.

On a sidenote thanks for doing this because cleaning categories up was on my todo list for months now. I wish I was taking notes on what was wrong so I could tell you all of it right now, but I didn't prepare 😿

File: data/json/items/tool/science.json (hunk 3 above)
Comment (Devengel) [reply data/json/items/tool/science.json:615]: put hydrogen & nitrogen tank in other, No uses for them while looking through HHG.

**Comment by Devengel** [comment]
This PR was originally meant for TOOL MAGAZINES so i'm not gonna do too much. i'm more of a mapgen person than items

**Comment by Devengel** [comment]
i think i've put most items i know of where appropriate? let me know if i missed anything. here is what changed so far.
image
image

**Review by ampersand55 — CHANGES_REQUESTED** [review]

File: data/json/items/tool/med.json (hunk 2 above)
Comment (Devengel) [reply data/json/items/tool/med.json:38]: Yeah, i think anesthesia deserves a rework
I placed it in drugs so it doesn't get lost in other categories. as its really just a glorified drug.

**Comment by Devengel** [comment]
Oh! batteries, how did i forget that, going to do them

**Comment by ampersand55** [comment]
It seems the the charges on hydrogen tank and nitrogen tank doesn't work. 

I propose we replace this in hydrogen/nitrogen tank, as it does nothing AFAICT:

```JSON
    "initial_charges": 24,
    "max_charges": 24,
```

With what oxygen cylinder and butane tank use:

```JSON
    "pocket_data": [
      {
        "pocket_type": "MAGAZINE",
        "rigid": true,
        "airtight": true,
        "ammo_restriction": { "hydrogen": 24 },
        "watertight": true
      }
    ],
    "flags": [ "NO_UNLOAD", "NO_RELOAD" ],
```

And add corresponding ammo types in ammo.json. 

Or, if you're feeling really ambitious, define all gasses in materials.json and use ` "pocket_type": "CONTAINER",`

**Comment by ampersand55** [comment]
Oh, just spotted something in the butane tank definition:

```JSON
 "explode_in_fire": true,
```

Would be very fitting for the hydrogen tank. Not sure what it does but I bet it's cool.

**Comment by Devengel** [comment]
Just tested it, does nothing. same with butane tank, they just get deleted.

**Comment by ampersand55** [comment]
They're just fluff item for now anyway. I just think it adds to immersion when they're loaded with something.

**Comment by Devengel** [comment]
image
image

**Comment by Karol1223** [comment]
> They're just fluff item for now anyway. I just think it adds to immersion when they're loaded with something.

*Are* they? I am 99% sure hydrogen tanks at least have a use in this weird diamond coating mechanic we have, though truth be told I don't know if that's still around.

**Comment by Devengel** [comment]
> > They're just fluff item for now anyway. I just think it adds to immersion when they're loaded with something.
> 
> _Are_ they? I am 99% sure hydrogen tanks at least have a use in this weird diamond coating mechanic we have, though truth be told I don't know if that's still around. But if it IS, be careful with changing them since it may be hardcoded.

Just tested it, CVD machine does not use hydrogen tanks. it uses hydrogen canisters, which contain solid hydrogen.

**Comment by ampersand55** [comment]
Seems the nitrogen tanks are filled with hydrogen.

**Comment by ampersand55** [comment]
I can see why one would mix up hydrogen in hydrogen tanks and hydrogen canisters. Perhaps rename `hydrogen canister` to `solid hydrogen canister` or similar?

The plasma guns also use hydrogen canisters as ammo.

**Pull request merged**